app/assets/javascripts/discourse/app/instance-initializers/auth-complete.js

```diff
@@ -140,9 +140,11 @@ export default {
               };
 
               if (siteSettings.experimental_full_page_login) {
-                router.transitionTo("signup").then((login) => {
+                router.transitionTo("signup").then((signup) => {
+                  const signupController =
+                    signup.controller || owner.lookup("controller:signup");
```

Comment:
When would there not be a controller returned from the transition?

Reply:
From my testing, when the origin is `/signup` and we do `router.transitionTo("signup")`, the controller is `undefined`. That's why it does work when signing up from `/login`.